jung-visualization/src/test/java/edu/uci/ics/jung/visualization/spatial/rtree/RTreeTest.java

```diff
@@ -227,11 +227,23 @@ private void testAreas(TreeNode rootNode) {
     Rectangle2D rootBounds = rootNode.getBounds();
     if (rootNode instanceof InnerNode) {
       InnerNode innerNode = (InnerNode) rootNode;
-      Assert.assertEquals(rootBounds, Node.union(innerNode.getChildren()));
+      Assert.assertTrue(closeEnough(rootBounds, Node.union(innerNode.getChildren())));
     }
 
     for (TreeNode rt : rootNode.getChildren()) {
       testAreas(rt);
     }
   }
+
+  private boolean closeEnough(Rectangle2D left, Rectangle2D right) {
+    return left.equals(right)
+        || (closeEnough(left.getMinX(), right.getMinX())
+            && closeEnough(left.getMinY(), right.getMinY())
+            && closeEnough(left.getMaxX(), right.getMaxX())
+            && closeEnough(left.getMaxY(), right.getMaxY()));
+  }
+
+  private boolean closeEnough(double left, double right) {
+    return Math.abs(left - right) < 0.001;
```

Comment:
Please define a constant for the tolerance.

Reply:
I'm delighted with your idea for getLocations too.